pkg/controllers/infrastatus/reconcile.go

```diff
@@ -0,0 +1,78 @@
+package infrastatus
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/go-logr/logr"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	configlister "github.com/openshift/client-go/config/listers/config/v1"
+)
+
+type InfraStatusReconciler struct {
+	Source     *configv1.Infrastructure
+	Client     configclient.Interface
+	KubeClient kubeclient.Interface
+	Lister     configlister.InfrastructureLister
+	Log        logr.Logger
+
+	m              sync.Mutex
+	hasSubresource *bool
+}
+
+func (r *InfraStatusReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	proceed, err := r.shouldReconcile()
+	if err != nil || !proceed {
+		return ctrl.Result{}, err
+	}
+	if req.Name != "cluster" {
+		return ctrl.Result{}, nil
+	}
+	r.Log.Info("Reconciling", "name", req.NamespacedName.String())
+	existing, err := r.Lister.Get(req.Name)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("cannot fetch infrastructure %s: %v", req.Name, err)
+	}
+	updateNeeded := false
+	updated := existing.DeepCopy()
+	r.Source.Status.DeepCopyInto(&updated.Status)
+
+	if !reflect.DeepEqual(updated.Status, existing.Status) {
+		updateNeeded = true
+	}
+
+	if !updateNeeded {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Couldn't you remove the updateNeeded var above and just simplify this?

```suggestion
	if reflect.DeepEqual(updated.Status, existing.Status) {
              return ctrl.Result{}, nil
	}
```